assets/src/block-editor/components/amp-preview.js

```diff
@@ -0,0 +1,283 @@
+		// https://html.spec.whatwg.org/multipage/interaction.html#dom-window-focus
+		this.previewWindow.focus();
+
+		// If we don't need to autosave the post before previewing, then we simply
+		// load the Preview URL in the Preview tab.
+		if ( ! this.props.isAutosaveable ) {
+			this.setPreviewWindowLink( event.target.href );
+			return;
+		}
+
+		// Request an autosave. This happens asynchronously and causes the component
+		// to update when finished.
+		if ( this.props.isDraft ) {
+			this.props.savePost( { isPreview: true } );
+		} else {
+			this.props.autosave( { isPreview: true } );
+		}
+
+		// Display a 'Generating preview' message in the Preview tab while we wait for the
+		// autosave to finish.
+		writeInterstitialMessage( this.previewWindow.document );
+	}
+
+	/**
+	 * Renders the component.
+	 */
+	render() {
+		const { previewLink, currentPostLink, isSaveable } = this.props;
+
+		// Link to the `?preview=true` URL if we have it, since this lets us see
+		// changes that were autosaved since the post was last published. Otherwise,
+		// just link to the post's URL.
+		const href = previewLink || currentPostLink;
+
+		return (
+			! errorMessages.length && (
+				<PluginPostStatusInfo>
+					<Button
+						isLarge
+						className="editor-post-preview"
+						href={ href }
+						target={ this.getWindowTarget() }
+						disabled={ ! isSaveable }
+						onClick={ this.openPreviewWindow }
+					>
+						{ __( 'AMP Preview', 'amp' ) }
+						<span className="screen-reader-text">
+							{
+								/* translators: accessibility text */
+								__( '(opens in a new tab)', 'amp' )
+							}
+						</span>
+						<DotTip tipId="core/editor.preview">
+							{ __( 'Click “Preview” to load a preview of this page in AMP, so you can make sure you’re happy with your blocks.', 'amp' ) }
+						</DotTip>
+					</Button>
+				</PluginPostStatusInfo>
+			)
+		);
+	}
+}
+
+AMPPreview.propTypes = {
+	autosave: PropTypes.bool.isRequired,
+	currentPostLink: PropTypes.func.isRequired,
+	postId: PropTypes.bool.isRequired,
+	previewLink: PropTypes.func.isRequired,
+	isAutosaveable: PropTypes.func.isRequired,
+	isDraft: PropTypes.func.isRequired,
+	isSaveable: PropTypes.func.isRequired,
+	isViewable: PropTypes.func.isRequired,
+	savePost: PropTypes.bool.isRequired,
+};
+
+export default compose( [
+	withSelect( ( select, { forcePreviewLink, forceIsAutosaveable } ) => {
+		const {
+			getCurrentPostId,
+			getCurrentPostAttribute,
+			getEditedPostAttribute,
+			isEditedPostSaveable,
+			isEditedPostAutosaveable,
+			getEditedPostPreviewLink,
+		} = select( 'core/editor' );
+		const {
+			getPostType,
+		} = select( 'core' );
+
+		const initialPreviewLink = getEditedPostPreviewLink();
+		const previewLink = initialPreviewLink ? addQueryArgs( initialPreviewLink, { amp: 1 } ) : undefined;
+		const postType = getPostType( getEditedPostAttribute( 'type' ) );
+
+		return {
+			postId: getCurrentPostId(),
+			currentPostLink: getCurrentPostAttribute( 'link' ),
```

Comment:
This should get the query var added to it as well.

Reply:
Sure, that's applied [here](https://github.com/ampproject/amp-wp/pull/3323/commits/7a574efb3c6247d655ed42d648dde6db0a77031e#diff-66f94e980255259f2c596579f2853515R272)